Live: use StreamingDataSource for live measurments

ryantxu opened the pull request:
extracted from #31010 to simplify that PR

ryantxu committed:
alternative

ryantxu committed:
more comments

leeoniya committed:
Merge branch 'master' into data-frame-json

leeoniya committed:
refactor & restructure

leeoniya committed:
fix DataFrameData interface

ryantxu committed:
add key and replace measurments

ryantxu committed:
add key and replace measurments

ryantxu committed:
comments

ryantxu committed:
Merge remote-tracking branch 'origin/master' into data-frame-json

ryantxu committed:
Merge remote-tracking branch 'origin/master' into data-frame-json

ryantxu committed:
values

ryantxu committed:
fix snapshot

ryantxu committed:
add alpha

ryantxu committed:
leon comments

ryantxu committed:
Merge remote-tracking branch 'origin/master' into data-frame-json

ryantxu committed:
Update packages/grafana-data/src/dataframe/DataFrameJSON.test.ts

ryantxu committed:
nan, inf, undefined

ryantxu committed:
Merge remote-tracking branch 'origin/master' into data-frame-json

ryantxu committed:
Merge remote-tracking branch 'origin/master' into data-frame-json

torkelo reviewed (approved):
Looks good, 

Is the plan to migrate from arrow base64 to DataFrameJSON later?

leeoniya commented:
yep

ryantxu committed:
merge master

ryantxu committed:
Merge remote-tracking branch 'origin/master' into measurment-collector

ryantxu committed:
write json structure

ryantxu commented on go.mod:
```diff
@@ -42,7 +42,7 @@ require (
 	github.com/grafana/grafana-aws-sdk v0.2.0
 	github.com/grafana/grafana-plugin-model v0.0.0-20190930120109-1fc953a61fb4
-	github.com/grafana/grafana-plugin-sdk-go v0.88.0
+	github.com/grafana/grafana-plugin-sdk-go v0.88.1-0.20210316010403-be8d0c26d32a
```
This includes the FrameToJSON function... we should use a tagged release before merging!

wbrowne committed:
move backend docs to their relevant folder (#32041)

Commit:
Chore: Collect elasticsearch version usage stats (#31787)

* Chore: Collect elasticsearch version usage stats

* Fix lint error

* use GetDataSources from sqlstore

* Apply review suggestions

* Return error if datasource type is not specified

* Update pkg/services/sqlstore/datasource.go

* fix undefined var

thisisobate committed:
Form: update story from knobs to controls (#32022)

* Form: update story from knobs to controls

* some code cleanup

papagian committed:
AlertingNG: Unified alerting API mock (#32040)

* AlertingNG: Alertmanager mock API

* AlertingNG: Remove permissions API routes

* Add example POST payloads

* Prometheus and testing mock API

marefr committed:
Data proxy: Fix encoded characters in URL path should be proxied encoded (#30597)

Fix encoded characters in URL path should be proxied as encoded in the data proxy.

Fixes #26870 
Fixes #31438

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

dependabot[bot] committed:
Chore(deps): Bump github.com/aws/aws-sdk-go from 1.37.30 to 1.37.32 (#32058)

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.37.30 to 1.37.32.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.37.30...v1.37.32)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

marefr committed:
Automation: Label pull requests based on changed files (#32039)

* Automation: Label pull requests based on changed files

* Update commands

* Updates for build-packaging

* fix path to backend architecture docs

dprokop committed:
GraphNG: Fix PlotLegend field display name being outdated (#32064)

thisisobate committed:
Legacy Input: updates story from knobs to control (#32059)

marefr committed:
Update pr commands glob paths (#32068)

aknuds1 committed:
PluginManager: Make Plugins, Renderer and DataSources non-global (#31866)

* PluginManager: Make Plugins and DataSources non-global

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix integration tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Replace outdated command

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* DashboardService: Ensure it gets constructed with necessary parameters

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix build

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* DashboardService: Ensure it gets constructed with necessary parameters

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Remove dead code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Remove FocusConvey

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Remove dead code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Undo interface changes

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Backend: Move tsdbifaces.RequestHandler to plugins.DataRequestHandler

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Rename to DataSourceCount

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Consolidate dashboard interfaces into one

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix dashboard integration tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

dsotirakis committed:
Downgrade to node:14.16.0-alpine3.13 (#32065)

gabor committed:
influxdb: raw-influxql editor: fix state-management bug (#32035)

* influxdb: raw-influxql editor: state-management bug fixed

* influxdb: simplified code

marefr committed:
Update pr commands glob paths (#32073)

marefr committed:
Update pr commands glob paths (#32074)

ryantxu committed:
Merge remote-tracking branch 'origin/master' into measurment-collector

ryantxu committed:
extract backend stuff

ryantxu commented on packages/grafana-runtime/src/measurement/collector.ts:
```diff
@@ -1,212 +1,84 @@
-    if (action === MeasurementAction.Clear) {
-      this.measurements.clear();
-      action = MeasurementAction.Append;
+  addBatch = (msg: MeasurementBatch) => {
```
@FZambia this is a temporary hack while we sync the frontend and backend implementations.  Currently on the backend, we have types for `DataFrameJSON`, but not a batch of them -- we are also missing `key` on DataFrameJSON and that will be required if we want to support multiple frames in one channel

ryantxu committed:
Merge remote-tracking branch 'origin/master' into measurment-collector

ryantxu committed:
update snapshots
